### ISSUE TITLE
Can't assign warehouse to a product

### ISSUE THREAD

**Issue by decimoseptimo** [mirumee/saleor-dashboard#852]
I'm experiencing an issue with `product > inventory > quantity`.

- I can update quantity or delete warehouse normally. 
- But adding a new warehouse just takes place on the UI, as it reverses after re-entering the product, the database does not save (`warehouse-stock` table).

How should i go about this? have you seen this before?
I'm using saleor-platform api worker. and a local production saleor-dashboard.

**Comment by shakenov-chinga** [comment]
Hi,
I'm having the same problem right now. I've checked graphql variables in browser console: the `addStocks` list is always empty.

**Comment by shakenov-chinga** [comment]
@decimoseptimo 
I've found a quick dirty fix!
In dashboard project folder `src/products/components/ProductUpdatePage/form.tsx` file at line 223 set `addStocks: stocks.data`.
Should look like this:
```
  const submitData: ProductUpdateSubmitData = {
    ...data,
    ...getAvailabilityData(data),
    ...getStocksData(product, stocks.data),
    ...getMetadata(data, isMetadataModified, isPrivateMetadataModified),
    addStocks: stocks.data,
    attributes: attributes.data
  };
```
The only problem left is that you can't add new warehouse upon existing warehouses. You will need to delete all warehouses, save, and then add old and new warehouses and save again.

I hope this will be fixed soon :smile: :+1:

**Comment by incorvia** [comment]
I was able to work around this by enabling variant support via the product type. I have not looked into this but it appears like stock is now mount to be staved at the variant level.. so either this UI for inventory at the product level was left behind and doesn't really work, meaning it's a bug.. or it's meant to be removed and it just hasn't yet? Someone may be able to chime in on what is the path forward here?

**Comment by mpita** [comment]
Hello everyone

I understand that it is used by variant. But from what I've seen at the code level, if you create a product without a variant and save it. This is registered with a stock directly to the product, there is when the error occurs.

Now I do not know if, you can no longer create a product without variants?

The solution indicated by @shakenov-chinga is fine, but in my opinion I don't think it is necessary to add addStocks: stocks.data to the function they invoke **... getStocksData (product, stocks.data),** returns {addStocks, removeStocks, updateStocks } so it is only necessary to remove the addStocks: [] in the getSubmitData function since it is overwriting what getStocksData returns. what I propose is something like this:

```
const getSubmitData = (): ProductUpdateSubmitData => ({
    ...data,
    ...getStocksData(product, stocks.data),
    ...getMetadata(data, isMetadataModified, isPrivateMetadataModified),
    attributes: attributes.data,
    attributesWithNewFileValue: attributesWithNewFileValue.data,
    description: description.current
 });
```

Best regards.
I love your project, and I congratulate the entire Mirumee team for the work they do.

**Comment by jwm0** [comment]
Hey, this should be now fixed in #939 and it's part of the next release.
